### ISSUE TITLE
Uploading ipa to testflight under linux

### ISSUE THREAD

**Issue by Vladeng** [fastlane/fastlane#15895]
### Question Checklist
- [x] Updated fastlane to the latest version
- [x] I read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I read [docs.fastlane.tools](https://docs.fastlane.tools)
- [x] I searched for [existing GitHub issues](https://github.com/fastlane/fastlane/issues)

Hi folks, 
I try to solve a very straight forward problem for my CI it's "upload ipa to testflight under linux".
I fount docker image with prebuild iTMSTransporter. I tried to do this stuff:
`sudo docker run --rm -itv `pwd`:/home/hex/test:rw --user root ineentho/fastlane-docker sh -c 'gem install fastlane && export FASTLANE_USER="ХХХ" && export FASTLANE_PASSWORD="ХХХ" &&  export FASTLANE_APPLE_APPLICATION_SPECIFIC_PASSWORD="ХХХ" && export DELIVER_USER="ХХХ" && export DELIVER_ITMSTRANSPORTER_ADDITIONAL_UPLOAD_PARAMETERS="-t DEV #" && fastlane pilot upload --verbose --ipa "../../hex/test/iracing.ipa"'`
Everythings fine.... almost in one of uploading somting blow up, and i cant figure it out. 
Maybe u have any solution or ides on how to solve this problem in the simplest way? 

```
Looking for App Store Connect Team with ID 1233985
INFO [2019-12-03 16:27:59.97]: Login successful
DEBUG [2019-12-03 16:27:59.99]: App identifier (com.miniit.iracing)
INFO [2019-12-03 16:28:03.22]: Ready to upload new build to TestFlight (App: 1454002895)...
DEBUG [2019-12-03 16:28:03.23]: App Platform (ios)
INFO [2019-12-03 16:28:03.87]: Wrote XML data to '/tmp/d20191203-5-zzneh0/1454002895.itmsp'
INFO [2019-12-03 16:28:03.87]: Fetching password for transporter from environment variable named `FASTLANE_APPLE_APPLICATION_SPECIFIC_PASSWORD`
INFO [2019-12-03 16:28:03.87]: Going to upload updated app to App Store Connect
INFO [2019-12-03 16:28:03.87]: This might take a few minutes. Please don't interrupt the script.
DEBUG [2019-12-03 16:28:03.87]: "/itms/bin/iTMSTransporter" -m upload -u hexware@gmail.com -p 'YourPassword' -f "/tmp/d20191203-5-zzneh0/1454002895.itmsp" -t DAV -t Signiant -k 100000
DEBUG [2019-12-03 16:29:20.28]: [Transporter]: INFO: Configuring logging...
DEBUG [2019-12-03 16:29:20.30]: [Transporter]: INFO: Logging level set to eXtreme
DEBUG [2019-12-03 16:29:20.30]: [Transporter]: INFO: Running inside of a Docker container. Bypassing automatic software update.
DEBUG [2019-12-03 16:29:20.52]: [Transporter]: INFO: Could not find transport type: Aspera
DEBUG [2019-12-03 16:29:20.52]: [Transporter]: INFO: Could not find transport type: DAV
DEBUG [2019-12-03 16:29:20.53]: [Transporter]: INFO: Could not find transport type: HTTP
DEBUG [2019-12-03 16:29:20.53]: [Transporter]: INFO: Could not find transport type: Signiant
ERROR [2019-12-03 16:29:21.96]: [Transporter Error Output]: An error occurred while trying to start the iTMSTransporter
WARN [2019-12-03 16:29:22.26]: [iTMSTransporter]        at com.apple.transporter.bootstrap.BootstrapperPhase2.runStarter(BootstrapperPhase2.java:317)

WARN [2019-12-03 16:29:22.26]: [iTMSTransporter]        at com.apple.transporter.bootstrap.BootstrapperPhase2.start(BootstrapperPhase2.java:130)

WARN [2019-12-03 16:29:22.26]: [iTMSTransporter]        at com.apple.transporter.bootstrap.BootstrapperPhase1.startOSGiFramework(BootstrapperPhase1.java:321)

WARN [2019-12-03 16:29:22.26]: [iTMSTransporter]        at com.apple.transporter.bootstrap.BootstrapperPhase1.bootstrap(BootstrapperPhase1.java:99)

WARN [2019-12-03 16:29:22.26]: [iTMSTransporter]        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)

WARN [2019-12-03 16:29:22.26]: [iTMSTransporter]        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)

WARN [2019-12-03 16:29:22.26]: [iTMSTransporter]        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)

WARN [2019-12-03 16:29:22.26]: [iTMSTransporter]        at java.lang.reflect.Method.invoke(Method.java:498)

WARN [2019-12-03 16:29:22.26]: [iTMSTransporter]        at com.apple.transporter.osgi.FelixOSGiBootstrapper.bootstrap(FelixOSGiBootstrapper.java:49)

WARN [2019-12-03 16:29:22.26]: [iTMSTransporter]        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)

WARN [2019-12-03 16:29:22.26]: [iTMSTransporter]        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)

WARN [2019-12-03 16:29:22.26]: [iTMSTransporter]        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)

WARN [2019-12-03 16:29:22.26]: [iTMSTransporter]        at java.lang.reflect.Method.invoke(Method.java:498)

WARN [2019-12-03 16:29:22.26]: [iTMSTransporter]        at com.apple.transporter.launcher.Application.start(Application.java:199)

WARN [2019-12-03 16:29:22.26]: [iTMSTransporter]        at com.apple.transporter.launcher.Application.main(Application.java:577)

INFO [2019-12-03 16:29:22.26]: iTunes Transporter output above ^
ERROR [2019-12-03 16:29:22.26]: An error occurred while trying to start the iTMSTransporter
The call to the iTMSTransporter completed with a non-zero exit status: 1. This indicates a failure.
ERROR [2019-12-03 16:29:22.26]: Could not download/upload from App Store Connect! It's probably related to your password or your internet connection.

```

**Comment by Vladeng** [comment]
<details><summary>✅ fastlane environment ✅</summary>

### Stack

| Key                         | Value                                                   |
| --------------------------- | ------------------------------------------------------- |
| OS                          | debian_based 9.9                                        |
| Ruby                        | 2.6.3                                                   |
| Bundler?                    | false                                                   |
| Git                         | git version 2.11.0                                      |
| Installation Source         | /usr/local/bundle/bin/fastlane                          |
| Host                        | Debian GNU/Linux 9 debian_based 9.9 (3.13.0-71-generic) |
| Ruby Lib Dir                | /usr/local/lib                                          |
| OpenSSL Version             | OpenSSL 1.1.0k  28 May 2019                             |
| Is contained                | false                                                   |
| Is homebrew                 | false                                                   |
| Is installed via Fabric.app | false                                                   |


### System Locale

| Variable | Value   |   |
| -------- | ------- | - |
| LANG     | C.UTF-8 | ✅ |
| LC_ALL   | C.UTF-8 | ✅ |
| LANGUAGE |         |   |


### fastlane files:

**No Fastfile found**


**No Appfile found**


### fastlane gems

| Gem      | Version | Update-Status |
| -------- | ------- | ------------- |
| fastlane | 2.140.0 | ✅ Up-To-Date  |


### Loaded fastlane plugins:

**No plugins Loaded**


<details><summary><b>Loaded gems</b></summary>

| Gem                       | Version      |
| ------------------------- | ------------ |
| did_you_mean              | 1.3.0        |
| slack-notifier            | 2.3.2        |
| atomos                    | 0.1.3        |
| CFPropertyList            | 3.0.0        |
| claide                    | 1.0.2        |
| colored2                  | 3.1.2        |
| nanaimo                   | 0.2.6        |
| xcodeproj                 | 1.14.0       |
| rouge                     | 2.0.7        |
| xcpretty                  | 0.3.0        |
| terminal-notifier         | 2.0.0        |
| unicode-display_width     | 1.6.0        |
| terminal-table            | 1.8.0        |
| plist                     | 3.5.0        |
| public_suffix             | 2.0.5        |
| addressable               | 2.6.0        |
| multipart-post            | 2.0.0        |
| word_wrap                 | 1.0.0        |
| tty-screen                | 0.7.0        |
| tty-cursor                | 0.7.0        |
| tty-spinner               | 0.9.1        |
| babosa                    | 1.0.2        |
| colored                   | 1.2          |
| highline                  | 1.7.10       |
| commander-fastlane        | 4.4.6        |
| excon                     | 0.71.1       |
| unf_ext                   | 0.0.7.6      |
| unf                       | 0.1.4        |
| domain_name               | 0.5.20190701 |
| http-cookie               | 1.0.3        |
| faraday-cookie_jar        | 0.0.6        |
| faraday                   | 0.17.3       |
| faraday_middleware        | 0.13.1       |
| fastimage                 | 2.1.5        |
| gh_inspector              | 1.1.3        |
| json                      | 2.1.0        |
| mini_magick               | 4.9.5        |
| multi_xml                 | 0.6.0        |
| rubyzip                   | 1.3.0        |
| security                  | 0.1.3        |
| xcpretty-travis-formatter | 1.0.0        |
| dotenv                    | 2.7.5        |
| bundler                   | 1.17.2       |
| naturally                 | 2.2.0        |
| simctl                    | 1.6.5        |
| jwt                       | 2.1.0        |
| uber                      | 0.1.0        |
| declarative               | 0.0.10       |
| declarative-option        | 0.1.0        |
| representable             | 3.0.4        |
| retriable                 | 3.1.2        |
| mini_mime                 | 1.0.2        |
| multi_json                | 1.13.1       |
| signet                    | 0.12.0       |
| memoist                   | 0.16.0       |
| os                        | 1.0.1        |
| googleauth                | 0.9.0        |
| httpclient                | 2.8.3        |
| google-api-client         | 0.36.4       |
| google-cloud-env          | 1.2.0        |
| google-cloud-core         | 1.3.0        |
| digest-crc                | 0.4.1        |
| google-cloud-storage      | 1.16.0       |
| emoji_regex               | 1.0.1        |
| forwardable               | 1.2.0        |
| logger                    | 1.3.0        |
| stringio                  | 0.0.2        |
| ipaddr                    | 1.2.2        |
| openssl                   | 2.1.2        |
| ostruct                   | 0.1.0        |
| strscan                   | 1.0.0        |
| date                      | 2.0.0        |
| fileutils                 | 1.1.0        |
| etc                       | 1.0.1        |
| io-console                | 0.4.7        |
| zlib                      | 1.0.0        |
| rexml                     | 3.1.9        |
| psych                     | 3.1.0        |
| mutex_m                   | 0.1.0        |
</details>


*generated on:* **2020-01-14**
</details>

**Comment by janpio** [comment]
> DEBUG [2019-12-03 16:29:20.52]: [Transporter]: INFO: Could not find transport type: Aspera
> DEBUG [2019-12-03 16:29:20.52]: [Transporter]: INFO: Could not find transport type: DAV
> DEBUG [2019-12-03 16:29:20.53]: [Transporter]: INFO: Could not find transport type: HTTP
> DEBUG [2019-12-03 16:29:20.53]: [Transporter]: INFO: Could not find transport type: Signiant

This looks like a broken iTMSTransporter install to me.

**Comment by Vladeng** [comment]
> > DEBUG [2019-12-03 16:29:20.52]: [Transporter]: INFO: Could not find transport type: Aspera
> > DEBUG [2019-12-03 16:29:20.52]: [Transporter]: INFO: Could not find transport type: DAV
> > DEBUG [2019-12-03 16:29:20.53]: [Transporter]: INFO: Could not find transport type: HTTP
> > DEBUG [2019-12-03 16:29:20.53]: [Transporter]: INFO: Could not find transport type: Signiant
> 
> This looks like a broken iTMSTransporter install to me.
thx for u answer but I don't know way to check it. As I tried to run only iTMSTransporter, I got:
`/itms/bin/iTMSTransporter
Downloading missing resources for version: 2.0.0
Analyzing remote bundle resource com.apple.its.epubcheck-runner...
  ... updated bundle successfully retrieved.
Analyzing remote bundle resource com.apple.its.epubtoolkit...
  ... updated bundle successfully retrieved.
Analyzing remote bundle resource com.apple.transporter.api...
  ... updated bundle successfully retrieved.
Analyzing remote bundle resource com.apple.transporter.aspera-linux-x64...
  ... updated bundle successfully retrieved.
Analyzing remote bundle resource com.apple.transporter.aspera-macos-x64...
  ... updated bundle successfully retrieved.
Analyzing remote bundle resource com.apple.transporter.aspera-transport...
  ... updated bundle successfully retrieved.
Analyzing remote bundle resource com.apple.transporter.aspera-win-x86...
  ... updated bundle successfully retrieved.
Analyzing remote bundle resource com.apple.transporter.commlink...
  ... updated bundle successfully retrieved.
Analyzing remote bundle resource com.apple.transporter.core...
  ... updated bundle successfully retrieved.
Analyzing remote bundle resource com.apple.transporter.core-transport...
  ... updated bundle successfully retrieved.
Analyzing remote bundle resource com.apple.transporter.dav-transport...
  ... updated bundle successfully retrieved.
Analyzing remote bundle resource com.apple.transporter.direct-san-transport...
  ... updated bundle successfully retrieved.
Analyzing remote bundle resource com.apple.transporter.http-transport...
  ... updated bundle successfully retrieved.
Analyzing remote bundle resource com.apple.transporter.itmstransporter...
  ... updated bundle successfully retrieved.
Analyzing remote bundle resource com.apple.transporter.jaxen-jdom...
  ... updated bundle successfully retrieved.
Analyzing remote bundle resource com.apple.transporter.launcher-util...
  ... updated bundle successfully retrieved.
Analyzing remote bundle resource com.apple.transporter.log4j2...
  ... updated bundle successfully retrieved.
Analyzing remote bundle resource com.apple.transporter.mediatoolkit...
  ... updated bundle successfully retrieved.
Analyzing remote bundle resource com.apple.transporter.resources...
  ... updated bundle successfully retrieved.
Analyzing remote bundle resource com.apple.transporter.signiant-transport...
  ... updated bundle successfully retrieved.
Analyzing remote bundle resource com.apple.transporter.softwaresupport...
  ... updated bundle successfully retrieved.
Analyzing remote bundle resource com.apple.transporter.starter...
  ... updated bundle successfully retrieved.
Analyzing remote bundle resource com.apple.transporter.updater...
  ... updated bundle successfully retrieved.
Analyzing remote bundle resource com.aspera.faspmanager-osgi...
  ... updated bundle successfully retrieved.
Analyzing remote bundle resource com.fasterxml.jackson.core.jackson-annotations...
  ... updated bundle successfully retrieved.
Analyzing remote bundle resource com.fasterxml.jackson.core.jackson-core...
  ... updated bundle successfully retrieved.
Analyzing remote bundle resource com.fasterxml.jackson.core.jackson-databind...
  ... updated bundle successfully retrieved.
Analyzing remote bundle resource com.fasterxml.jackson.dataformat.jackson-dataformat-xml...
  ... updated bundle successfully retrieved.
Analyzing remote bundle resource com.fasterxml.jackson.module.jackson-module-jaxb-annotations...
  ... updated bundle successfully retrieved.
Analyzing remote bundle resource com.fasterxml.woodstox.woodstox-core...
  ... updated bundle successfully retrieved.
Analyzing remote bundle resource com.github.oxo42.stateless4j...
  ... updated bundle successfully retrieved.
Analyzing remote bundle resource com.google.gson...
  ... updated bundle successfully retrieved.
Analyzing remote bundle resource com.google.guava...
  ... updated bundle successfully retrieved.
Analyzing remote bundle resource com.google.http-client.google-http-client...
  ... updated bundle successfully retrieved.
Analyzing remote bundle resource com.googlecode.plist.dd-plist...
  ... updated bundle successfully retrieved.
Analyzing remote bundle resource com.sun.activation.javax.activation...
  ... updated bundle successfully retrieved.
Analyzing remote bundle resource com.sun.istack.commons-runtime...
  ... updated bundle successfully retrieved.
Analyzing remote bundle resource com.sun.xml.bind.jaxb-impl...
  ... updated bundle successfully retrieved.
Analyzing remote bundle resource jaxb-api...
  ... updated bundle successfully retrieved.
Analyzing remote bundle resource json...
  ... updated bundle successfully retrieved.
Analyzing remote bundle resource org.apache.commons.cli...
  ... updated bundle successfully retrieved.
Analyzing remote bundle resource org.apache.commons.exec...
  ... updated bundle successfully retrieved.
Analyzing remote bundle resource org.apache.felix.configadmin...
  ... updated bundle successfully retrieved.
Analyzing remote bundle resource org.apache.felix.main...
  ... updated bundle successfully retrieved.
Analyzing remote bundle resource org.apache.felix.scr...
  ... updated bundle successfully retrieved.
Analyzing remote bundle resource org.apache.httpcomponents.httpclient...
  ... updated bundle successfully retrieved.
Analyzing remote bundle resource org.apache.httpcomponents.httpcore...
  ... updated bundle successfully retrieved.
Analyzing remote bundle resource org.apache.logging.log4j.api...
  ... updated bundle successfully retrieved.
Analyzing remote bundle resource org.apache.logging.log4j.core...
  ... updated bundle successfully retrieved.
Analyzing remote bundle resource org.apache.logging.log4j.jcl...
  ... updated bundle successfully retrieved.
Analyzing remote bundle resource org.apache.logging.log4j.slf4j-impl...
  ... updated bundle successfully retrieved.
Analyzing remote bundle resource org.apache.tika.core...
  ... updated bundle successfully retrieved.
Analyzing remote bundle resource org.apache.tika.parsers...
  ... updated bundle successfully retrieved.
Analyzing remote bundle resource org.bitbucket.b_c.jose4j...
  ... updated bundle successfully retrieved.
Analyzing remote bundle resource org.jsoup...
  ... updated bundle successfully retrieved.
Analyzing remote bundle resource org.jsr-305...
  ... updated bundle successfully retrieved.
Analyzing remote bundle resource org.osgi.impl.bundle.repoindex.lib...
  ... updated bundle successfully retrieved.
Analyzing remote bundle resource org.xerial.sqlite-jdbc...
  ... updated bundle successfully retrieved.
Analyzing remote bundle resource stax2-api...
  ... updated bundle successfully retrieved.
[2020-01-20 11:15:11 UTC] <main>  INFO: Configuring logging...
[2020-01-20 11:15:11 UTC] <main>  INFO: Logging level set to eXtreme
[2020-01-20 11:15:11 UTC] <main>  INFO: Running inside of a Docker container. Bypassing automatic software update.
[2020-01-20 11:15:11 UTC] <main>  INFO: Could not find transport type: Aspera
[2020-01-20 11:15:11 UTC] <main>  INFO: Could not find transport type: DAV
[2020-01-20 11:15:11 UTC] <main>  INFO: Could not find transport type: HTTP
[2020-01-20 11:15:11 UTC] <main>  INFO: Could not find transport type: Signiant
[2020-01-20 11:15:13 UTC] <main> ERROR: An error occurred while trying to start the iTMSTransporter
java.lang.NullPointerException: serviceReferences parameter cannot be null
        at com.apple.transporter.util.ConfigurationUtil.getServiceReference(ConfigurationUtil.java:184)
        at com.apple.transporter.osgi.TransporterService.start(TransporterService.java:50)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:498)
        at com.apple.transporter.starter.TransporterStarter.runService(TransporterStarter.java:486)
        at com.apple.transporter.starter.TransporterStarter.runTransporterService(TransporterStarter.java:450)
        at com.apple.transporter.starter.TransporterStarter.start(TransporterStarter.java:188)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:498)
        at com.apple.transporter.bootstrap.BootstrapperPhase2.runStartable(BootstrapperPhase2.java:355)
        at com.apple.transporter.bootstrap.BootstrapperPhase2.runStarter(BootstrapperPhase2.java:317)
        at com.apple.transporter.bootstrap.BootstrapperPhase2.start(BootstrapperPhase2.java:130)
        at com.apple.transporter.bootstrap.BootstrapperPhase1.startOSGiFramework(BootstrapperPhase1.java:321)
        at com.apple.transporter.bootstrap.BootstrapperPhase1.bootstrap(BootstrapperPhase1.java:99)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:498)
        at com.apple.transporter.osgi.FelixOSGiBootstrapper.bootstrap(FelixOSGiBootstrapper.java:49)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:498)
        at com.apple.transporter.launcher.Application.start(Application.java:199)
        at com.apple.transporter.launcher.Application.main(Application.java:577)
`

In dokerfile it looks very simple 
`    && curl https://itunesconnect.apple.com/transporter/1.9.8/iTMSTransporterToolInstaller_1.9.8.exe > installer.exe \
    && 7z x installer.exe -oitms \
    && chmod +x itms/bin/iTMSTransporter` i got only one binary file.
